### ISSUE TITLE
GPLv3 infects the dependent codebase

### ISSUE THREAD

**Issue by HMaker** [ltworf/typedload#156]
Is it possible to change the license to one that does not requires the dependent codebase be open sourced?

**Comment by HMaker** [comment]
Nevermind, there is a better alternative [dataclasses-json](https://github.com/lidatong/dataclasses-json)

**Comment by ltworf** [comment]
> Is it possible to change the license to one that does not requires the dependent codebase be open sourced?

You need the freedom to take freedom of others away? I'd say MIT license has a bug.

> Nevermind, there is a better MIT licensed alternative dataclasses-json

It is MIT licensed, however it:

* requires to annotate all the classes
* is not extensible
* and doesn't support Union
* has dependencies

so it is "better" only in the sense of allowing you to take freedom away.

**Comment by HMaker** [comment]
Not all codebase can be open sourced, specially application specific code, that's why licenses that requires dependent codebase to be open sourced are being dropped. LGPL was created to overcome this problem, why not use it? dataclasses-json has better because I can use that, I am sure most of programs developed can't be open sourced. Also, MIT-like licenses does not take freedom of anyone, you were free to license your code under GPL, I am free to decide whatever I do with me own code too, but GPL does not allow me to _not_ open source my code.

**Comment by ltworf** [comment]
> Not all codebase can be open sourced

Debatable.

> specially application specific code

Remember that GPL doesn't impose redistribution. If you do some software that you use internally and do not distribute to users, GPL doesn't impose you to distribute it and to distribute your changes to GPL covered software.

> that's why licenses that requires dependent codebase to be open sourced are being dropped. 

By whom? By you, sure. I can't envision that gcc, linux, bash will become MIT anytime soon.

> LGPL was created to overcome this problem, why not use it?

LGPL is a compromise, I've only adopted it to be able to use this project at my own workplace, where I've written a certain amount of commits to this.

> dataclasses-json is better because I can use that, I am sure most of programs developed can't be open sourced.

Again… debatable, but of course use whatever you want.

> Also, MIT-like licenses does not take freedom of anyone

They give freedom to developers to take freedom away from their users. Users are "someone".

> but GPL does not allow me to not open source my code.

Which is exactly the point of using GPL in the first place…

I make this project in my free time. If you take it for your own gain, I will have worked for free for you. Why would I do that?

If I make this project in my free time and I help other free software projects, I have worked for free to help the world (and myself, since I live on the world).

Lots of developers of small projects are willing to work for free for your gains. I can't see why they do that. But I don't understand why you are getting so worked up about not being able to exploit my work to produce something that lacks freedom and only benefits you.
